workers: fix typo (#16220)
cloudflare/cloudflare-docs: src/content/docs/workflows/reference/limits.mdx

---
pcx_content_type: concept
title: Limits
sidebar:
  order: 4

---

import { Render } from "~/components"

> Limits that apply to authoring, deploying and running Workflows are detailed below. 

Reviewer: ```suggestion
Limits that apply to authoring, deploying, and running Workflows are detailed below.
```

Many limits are inherited from those applied to Workers scripts and as documented in the [Workers limits](/workers/platform/limits/) documentation.

> | Feature                                   | Workers Free            | Workers Paid          |

Reviewer 2: We are missing the Wall time limite here, which should be 15 minutes per lifetime (limited by alarm timeout)

Author: For Free?

| ----------------------------------------- | ----------------------- | --------------------- |
| Workflow class definitions per script     | 1MB max script size per [Worker size limits](/workers/platform/limits/#account-plan-limits) | 10MB max script size per [Worker size limits](/workers/platform/limits/#account-plan-limits)
| Total scripts per account                 | 100 | 500 (shared with [Worker script limits](/workers/platform/limits/#account-plan-limits) |
| Compute time per Workflow                 | 10 seconds | 30 seconds of [active CPU time](/workers/platform/limits/#cpu-time) |
| Duration (wall clock) per `step`          | Unlimited | Unlimited - e.g. waiting on network I/O calls or querying a database |
| Maximum persisted state per step          | 1MiB (2^20 bytes) | 1MiB (2^20 bytes) | 
| Maximum state that can be persisted per Workflow instance | 100MB | 1GB | 

> | Maximum `step.sleep` duration             | 365 days (1 year) <beta>beta</beta> |

Reviewer: ```suggestion
| Maximum `step.sleep` duration             | 365 days (1 year) [^1] |
```

> | Maximum steps per Workflow                | 256 <beta>beta</beta> |

Reviewer: ```suggestion
| Maximum steps per Workflow                | 256 [^1] |
```

| Maximum Workflow executions               | 100,000 per day [shared with Workers daily limit](/workers/platform/limits/#worker-limits) | Unlimited |
| Concurrent Workflow instances (executions) | 25 | 1,000 <beta>beta</beta> |

> | Retention limit for completed Workflow state | 3 days | 30 days <sup>1</sup> |

Reviewer: ```suggestion
| Retention limit for completed Workflow state | 3 days | 30 days [^2] |
```

> <beta>beta</beta> This limit will be reviewed and revised during the open beta for Workflows, and we'll update this page and our changelog.

Reviewer: ```suggestion
[^1]: This limit will be reviewed and revised during the open beta for Workflows, and we'll update this page and our changelog.
```

> <sup>1</sup> Workflow state and logs will be retained for 3 days on the Workers Free plan and 30 days on the Workers Paid plan.

Reviewer: ```suggestion
[^2]: Workflow state and logs will be retained for 3 days on the Workers Free plan and 30 days on the Workers Paid plan.
```
